### PR TITLE
Add a basic server with logging, config management

### PR DESCRIPTION
**Pull request by arkn98**
This PR does the following things:

- Sets up a basic server that returns a welcome message.
- Adds a centralized logger to be used by all other packages.
- Sets up config management using `viper` to read config from files and also from `ENV`.

### CODE REVIEW

File: cmd/adeia-api/main.go [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+	"adeia-api/internal/config"
+	log "adeia-api/internal/logger"
+	"adeia-api/internal/server"
+	"fmt"
+	"os"
+)
 
-func Hello() string {
-	return "Hello, world\n"
+func onError(msg string, err error) {
```
Comment (dyaneshvarun) [inline cmd/adeia-api/main.go:11]: Move it into Utils in the next PR.

File: internal/logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		err = nil
+
+		// parse log level
+		level, e := parseLevel(config.GetString("logger.level"))
+		if e != nil {
+			err = e
+			return
+		}
+
+		// TODO: switch to custom config
```
Comment (dyaneshvarun) [inline internal/logger/logger.go:54]: Are these TODO's done ?!

File: cmd/adeia-api/main.go (hunk 1 above)
Comment (arkn98) [reply cmd/adeia-api/main.go:11]: Ok sure!

File: internal/logger/logger.go (hunk 2 above)
Comment (arkn98) [reply internal/logger/logger.go:54]: Nope. We currently do not have any requirement for using a custom config. But we should be doing it in the course of time.

**Comment by arkn98** [comment]
@dyaneshvarun Raised a new PR (#7) with the requested changes.

File: internal/config/config.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+// LoadConf loads YAML from file specified by EnvConfPathKey into viper.
+// The file must be a readable file containing valid YAML.
+func LoadConf() error {
+	err := errors.New("config already loaded")
+
+	initConf.Do(func() {
+		err = nil
+
+		confPath := getEnv(EnvConfPathKey, "./config/config.yaml")
```
Comment (dyaneshvarun) [inline internal/config/config.go:28]: This doesnt work as such dood. Should be discussed.

**Pull request merged**